cmds/core/dd/dd_test.go

```diff
@@ -314,22 +311,37 @@ func TestInFile(t *testing.T) {
 		},
 	}
 
-	err := ioutil.WriteFile("datafile", []byte("ABCDEFG"), 0644)
-	if err != nil {
-		t.Errorf("unable to mockup file: %v", err)
-	}
-	defer os.Remove("datafile")
+	p, cleanup := setupDatafile(t, "datafile")
+	defer cleanup()
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, err = inFile(tt.filename, tt.outputBytes, tt.seek, tt.count)
+			_, err := inFile(p, tt.outputBytes, tt.seek, tt.count)
 			if err != nil && !tt.wantErr {
 				t.Errorf("outFile failed with %v", err)
 			}
 		})
 	}
 }
 
+func setupDatafile(t *testing.T, name string) (string, func()) {
+	t.Helper()
+
+	testDir, err := ioutil.TempDir("", "dd")
```

Comment:
Could this be https://pkg.go.dev/testing#B.TempDir ?

Reply:
#B.TempDir seems to call on a B object, does not seem like a good reason to use `testing.B` here.

Reply:
What about T.TempDir?

Reply:
Here, if we use `testing.T.TempDir`,  it will achieve the same effect, just one cant supply a suffix as a tmpdir name like `ioutil.TempDir("", "dd")` provides. Ok to stay as is, or change it to use testing.T if that proves to be better ^^

Reply:
Yes, it's not a blocker though.

Reply:
ack.